src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -120,7 +121,11 @@ public void setFilePath(String filePath) {
                             logger.warn("Invalid override property: " + valueObj);
                         }
                     } else {
-                        addConfig(result, "", configKey, valueObj);
+                        if (valueObj instanceof Map) {
```

Comment:
Could you check "valueObj instanceof Map" inside addConfig, so we keep the code here simple without checking value type. All details will be handled by addConfig() method.